VariationalBayes/test_exponential_families.py

```diff
@@ -103,6 +109,66 @@ def get_r_matrix(mat):
             ef.expected_ljk_prior(lkj_param, df, v),
             atol=moment_tolerance)
 
+    def test_logitnormal_moments(self):
+        # log normal parameters
+        lognorm_means = np.random.random((5, 3)) # should work for arrays now
+        lognorm_infos = np.random.random((5, 3))**2 + 1
+        alpha = 2 # dp parameter
+
+        # draw samples
+        num_draws = 10**5
+        samples = np.random.normal(lognorm_means,
+                        1/np.sqrt(lognorm_infos), size = (num_draws, 5, 3))
+        logit_norm_samples = sp.special.expit(samples)
+
+        # test lognormal means
+        np_test.assert_allclose(
+            np.mean(logit_norm_samples, axis = 0),
+            ef.get_e_logitnormal(
+                lognorm_means, lognorm_infos, GH_LOC, GH_WEIGHTS),
+            atol = 3 * np.std(logit_norm_samples) / np.sqrt(num_draws))
+            # I think the integral can only get to be so accurate; it seems
+            # to be good up to three decimal points.
+
+        # test Elog(x) and Elog(1-x)
+        log_logistic_norm = np.mean(np.log(logit_norm_samples), axis = 0)
+        log_1m_logistic_norm = np.mean(np.log(1 - logit_norm_samples), axis = 0)
+
+        tol1 = 3 * np.std(np.log(logit_norm_samples))/ np.sqrt(num_draws)
+        tol2 = 3 * np.std(np.log(1 - logit_norm_samples))/ np.sqrt(num_draws)
+
+        np_test.assert_allclose(
+            log_logistic_norm,
+            ef.get_e_log_logitnormal(
+                lognorm_means, lognorm_infos, GH_LOC, GH_WEIGHTS)[0],
+            atol = tol1)
+
+        np_test.assert_allclose(
+            log_1m_logistic_norm,
+            ef.get_e_log_logitnormal(
+                        lognorm_means, lognorm_infos, GH_LOC, GH_WEIGHTS)[1],
+            atol = tol2)
+
+        # test prior
+        prior_samples = np.mean((alpha - 1) *
+                            np.log(1 - logit_norm_samples), axis = 0)
+        tol3 = 3 * np.std((alpha - 1) * np.log(1 - logit_norm_samples)) \
+                    /np.sqrt(num_draws)
+        np_test.assert_allclose(
+            prior_samples,
+            ef.get_e_dp_prior_logitnorm_approx(
+                        alpha, lognorm_means, lognorm_infos, GH_LOC, GH_WEIGHTS),
+            atol = tol3)
+
+        # Was this what you meant when you wanted to check the autodiff?
```

Comment:
You can make comments in github, you know.  :)  You should delete these comments in the code before submitting.